macrosynergy/management/dq.py

```diff
@@ -281,218 +313,60 @@ def check_connection(self) -> bool:
-            '/groups/search' with params of keywords.
-
-        :return: JSON dictionary object with result of query
-        :rtype: <str>
-
-        """
-        if keywords is not None:
-            assert isinstance(keywords, str)
-            results = self._fetch(endpoint="/groups/search",
-                                  params={"keywords": keywords})
-        else:
-            results = self._fetch()
-
-        if self.debug:
-            print("\nCheck data set:")
-            print("Max number of groups (item):",
-                  max(map(lambda x: x["item"], results)))
-            print("Premium content:",
-                  any(map(lambda x: x["premium"], results)))
-            print("Providers:",
-                  np.unique(list(map(lambda x: x["provider"], results))))
-            # TODO "FX" in group-id?
-            print("Group ID (FX):",
-                  list(filter(lambda y: y[:2] == "FX" or y[:3] == "CFX",
-                              map(lambda x: x["group-id"], results))))
-            print("Group ID (FX):",
-                  list(filter(lambda y: "FX" in y,
-                              map(lambda x: x["group-id"], results))))
-            print("Athena FX:",
-                  list(filter(lambda y:
-                              y["provider"] == "ATHENA FX", results)))
-            print(list(filter(lambda x:
-                              x["group-id"] == "FXO_SP", results)))
-
-        return results
+    def _request(self, endpoint: str, tickers: List[str], params: dict,
+                 start_date: str = None, end_date: str = None,
+                 calendar: str = "CAL_ALLDAYS", frequency: str = "FREQ_DAY",
+                 conversion: str = "CONV_LASTBUS_ABS",
+                 nan_treatment: str = "NA_NOTHING"):
 
-    def get_instruments(self, group_id: str, keywords: str = None):
-        """Get all instruments within a group.
+        params_ = {"format": "JSON", "start-date": start_date, "end-date": end_date,
+                   "calendar": calendar, "frequency": frequency, "conversion":
+                   conversion, "nan_treatment": nan_treatment}
+        params.update(params_)
 
-        :param group_id: string denoting the group-id
-            for which to get all instruments.
-        :param keywords: string with keywords for search.
-            Default is None with endpoint '/group/instruments',
-        but if not None call '/group/instruments/search'
-        :return: JSON dictionary object with result of query
-        """
+        no_tickers = len(tickers)
+        iterations = ceil(no_tickers / 20)
+        remainder = no_tickers % 20
 
-        if keywords is not None:
-            results = self._fetch(endpoint="/group/instruments/search",
-                                  select="instruments",
-                                  params={"group-id": group_id,
-                                          "keywords": keywords})
+        tick_list_compr = []
```

Comment:
Simplification:
```python
tick_list_compr = [tickers[i * 20:min((i * 20) + 20,len(tickers)+1) for ii in range(iterations)]
```

Reply:
Will make the changes. Thanks for the recommendation.

Reply:
Change made.